lib/ZnapZend/ZFS.pm

```
@@ -122,6 +123,30 @@ sub snapshotExists {
     return grep { $snapshot eq $_ } @snapshots;
 }
 
+sub createDataSet {
+    my $self = shift;
+    my $dataSet = shift;
+    my $remote;
+
+    #just in case if someone aks to check '';
+    return 0 if !$dataSet;
+
```

If copied that one from the other functions, so you want to throw an Exception? :+1: